speclang_modern.tex

```diff
@@ -1060,6 +1060,30 @@ \subsection{Simple function contracts}
 effects. See Section~\ref{sec:multiplecontracts} for more details on
 default status of clauses.
 
+\subsubsection{Semantics of frame conditions}
+\label{sec:writesSemantics}
+It is worth pointing out that there are different treatments of
+frame conditions (\lstinline|assigns| statements) in various specification languages.
+The frame condition can follow either \emph{writes} semantics or \emph{modifies} semantics.
+\begin{itemize}
+	\item Under \lstinline|writes| (or assigns) semantics, only those memory locations listed in a frame condition may be \emph{written to}, that is, only those locations may be the target of an assignment statement or listed in the frame condition of a called function. This is true whether or not the value of the memory location changes.
+	\item Under \lstinline|modifies| semantics, a memory location may be written to, as long as the value is restored (that is, not modified) by the end of the scope of the function contract. Under this semantics, a frame condition is a requirement on the relationship between two states --- any memory location not a member of the frame condition must have the same value in its pre-state and its post-state.
+\end{itemize}
+Confusion can arise because the words \emph{assigns} and \emph{modifies} are sometimes used interchangeably. In particular, \textbf{ACSL \ifCPP{and \NAME} \iftoggle{isCPP}{use}{uses}
+	\emph{modifies} semantics, even though the frame condition is introduced by the \emph{assigns} keyword}.\footnote{For comparison, JML and the OpenJML tool define frame conditions to have write semantics but use the keywords assigns and modifies interchangeably; however, the KeY tool for JML implements modifies semantics. Ada/SPARK's data flow contracts effectively encode write semantics.}
+
+\subsubsection{\texorpdfstring{\lstinline|check|}{check} and \texorpdfstring{\lstinline|admit|}{admit} clauses}\label{sec:check-admit-clauses}
+
+As presented above, in their basic acceptions, \lstinline|requires|
+and \lstinline|ensures| clauses both must be checked when control reaches the
+corresponding point, and can be assumed to hold to continue the analysis. However
+if only one of these actions, a {\it clause-kind} modifier can be used.
```

Comment:
About *However... used*, this sentence seems incomprehensible to me.

Reply:
```suggestion
if only one of these actions is needed, a {\it clause-kind} modifier can be used.
```